src/Renderer/ThreeExtended/FeaturesUtils.js

```diff
@@ -0,0 +1,122 @@
+function pointIsOverLine(point, linePoints, precision) {
+    const x0 = point._values[0];
+    const y0 = point._values[1];
+    // in first j is last point of line and then it'll be (i - 1)
+    for (var i = 0, j = linePoints.length - 1; i < linePoints.length; j = i++) {
+        const x1 = linePoints[i]._values[0];
+        const y1 = linePoints[i]._values[1];
+        const x2 = linePoints[j]._values[0];
+        const y2 = linePoints[j]._values[1];
+
+        const Xp = x0 - x1;
+        const Yp = y0 - y1;
+
+        const x21 = x2 - x1;
+        const y21 = y2 - y1;
+        const n = Math.sqrt(x21 * x21 + y21 * y21);
+        const scalar = (Xp * x21 + Yp * y21) / n;
+
+        if (scalar >= -precision && scalar <= n + precision) {
```

Comment:
You still didn't explain the algorithm used.
But this line is weirdly non-symetric (`-precision` on the left but `n + precision` on the right). Are you sure it's correct?

Reply:
I add comment